graphdatascience/tests/integrationV2/arrow_client/v2/test_gds_arrow_client.py

```diff
@@ -0,0 +1,134 @@
+def session_connection(
+    network: Network, password_dir: Path, logs_dir: Path
+) -> Generator[GdsSessionConnectionInfo, None, None]:
+    yield from start_session(logs_dir, network, password_dir)
+
+
+@pytest.fixture(scope="package")
+def arrow_client(session_connection: GdsSessionConnectionInfo) -> AuthenticatedArrowClient:
+    return create_arrow_client(session_connection)
+
+
+@pytest.fixture(scope="package")
+def gds_arrow_client(arrow_client: AuthenticatedArrowClient) -> GdsArrowClient:
+    return GdsArrowClient(arrow_client)
+
+
+@pytest.fixture
+def sample_graph(arrow_client: AuthenticatedArrowClient) -> Generator[GraphV2, None, None]:
+    gdl = """
+        CREATE
+        (a: Node:Foo {prop1: 1, prop2: 42.0}),
+        (b: Node {prop1: 2, prop2: 43.0}),
+        (c: Node:Foo {prop1: 3, prop2: 44.0}),
+        
+        (a)-[:REL {relX: 1, relY: 42}]->(b),
+        (b)-[:REL {relX: 2, relY: 43}]->(c),
+        (c)-[:REL2 {relX: 1, relY: 2}]->(a),
+    """
+
+    with create_graph(arrow_client, "g", gdl) as G:
+        yield G
+
+
+def test_stream_node_label(gds_arrow_client: GdsArrowClient, sample_graph: GraphV2) -> None:
+    job_id = gds_arrow_client.get_nodes(sample_graph.name(), node_filter="n.prop1 > 1")
+    result = gds_arrow_client.stream(sample_graph.name(), job_id)
+
+    assert ["nodeId"] == list(result.columns)
+    assert len(result) == 2
+
+
+def test_stream_node_properties(gds_arrow_client: GdsArrowClient, sample_graph: GraphV2) -> None:
+    job_id = gds_arrow_client.get_node_properties(sample_graph.name(), node_properties=["prop1", "prop2"])
+    result = gds_arrow_client.stream(sample_graph.name(), job_id)
+
+    assert len(result) == 3
+    assert "nodeId" in result.columns
+    assert "prop1" in result.columns
+    assert "prop2" in result.columns
+    assert set(result["prop1"].tolist()) == {1, 2, 3}
+    assert set(result["prop2"].tolist()) == {42.0, 43.0, 44.0}
+
+
+def test_stream_relationship_properties(gds_arrow_client: GdsArrowClient, sample_graph: GraphV2) -> None:
+    job_id = gds_arrow_client.get_relationships(sample_graph.name(), ["REL"], relationship_properties=["relX", "relY"])
+    result = gds_arrow_client.stream(sample_graph.name(), job_id)
+
+    assert len(result) == 2
+    assert "sourceNodeId" in result.columns
+    assert "targetNodeId" in result.columns
+    assert "relationshipType" in result.columns
+    assert "REL" in result["relationshipType"].tolist()
+    assert "relX" in result.columns
+    assert "relY" in result.columns
+    assert list(result["relX"].tolist()) == [1.0, 2.0]
+    assert list(result["relY"].tolist()) == [42.0, 43.0]
+
+
+def test_project_from_triplets(arrow_client: AuthenticatedArrowClient, gds_arrow_client: GdsArrowClient) -> None:
+    df = pd.DataFrame(
+        {"sourceNode": np.array([1, 2, 3], dtype=np.int64), "targetNode": np.array([4, 5, 6], dtype=np.int64)}
+    )
+
+    job_id = gds_arrow_client.create_graph_from_triplets("triplets")
```

Comment:
just a thought for later.
we could return a context manager from the call here and allow:

```
with gds_arrow_client.create_graph_from_triplets("triplets") as job_id:
     gds_arrow_client.upload_triplets(job_id, df)
```

this would automate aborting on exceptions and calling done in the end